elide-core/src/test/java/com/yahoo/elide/core/UpdateOnCreateTest.java

```diff
@@ -0,0 +1,232 @@
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.core;
+
+import com.yahoo.elide.core.exceptions.ForbiddenAccessException;
+import com.yahoo.elide.security.User;
+import com.yahoo.elide.utils.coerce.CoerceUtil;
+import example.Author;
+import example.Book;
+import example.UpdateAndCreate;
+import org.mockito.Answers;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.Test;
+
+import java.io.Serializable;
+import java.util.Optional;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class UpdateOnCreateTest extends PersistenceResourceTestSetup {
+
+    private RequestScope userOneScope;
+    private RequestScope userTwoScope;
+    private RequestScope userThreeScope;
+    private RequestScope userFourScope;
+
+    @BeforeTest
+    public void init() {
+        dictionary.bindEntity(Author.class);
+        dictionary.bindEntity(Book.class);
+        dictionary.bindEntity(UpdateAndCreate.class);
+
+        UpdateAndCreate updateAndCreateNewObject = new UpdateAndCreate();
+        updateAndCreateNewObject.setId(1L);
+        UpdateAndCreate updateAndCreateExistingObject = new UpdateAndCreate();
+        updateAndCreateExistingObject.setId(2L);
+        Book book = new Book();
+        Author author = new Author();
+
+        DataStoreTransaction tx = mock(DataStoreTransaction.class, Answers.CALLS_REAL_METHODS);
+
+        User userOne = new User(1);
+        userOneScope = new RequestScope(null, null, tx, userOne, null, elideSettings, false);
+        User userTwo = new User(2);
+        userTwoScope = new RequestScope(null, null, tx, userTwo, null, elideSettings, false);
+        User userThree = new User(3);
+        userThreeScope = new RequestScope(null, null, tx, userThree, null, elideSettings, false);
+        User userFour = new User(4);
+        userFourScope = new RequestScope(null, null, tx, userFour, null, elideSettings, false);
+
+        when(tx.createNewObject(UpdateAndCreate.class)).thenReturn(updateAndCreateNewObject);
+        when(tx.loadObject(eq(UpdateAndCreate.class),
+                eq((Serializable) CoerceUtil.coerce(1, Long.class)),
+                eq(Optional.empty()),
+                any(RequestScope.class)
+        )).thenReturn(updateAndCreateExistingObject);
+        when(tx.loadObject(eq(Book.class),
+                eq((Serializable) CoerceUtil.coerce(1, Long.class)),
+                eq(Optional.empty()),
+                any(RequestScope.class)
+        )).thenReturn(book);
+        when(tx.loadObject(eq(Author.class),
+                eq((Serializable) CoerceUtil.coerce(1, Long.class)),
+                eq(Optional.empty()),
+                any(RequestScope.class)
+        )).thenReturn(author);
+    }
+
+    //----------------------------------------- ** Entity Creation ** -------------------------------------------------
```

Comment:
Can you add a few more comments in each section explaining the purpose of the group of tests.